llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -489,6 +489,10 @@ class TargetLoweringBase {
     return true;
   }
 
+  /// Returns true if a floating-point load or store can be replaced with an
+  /// equivalent integer load or store without negatively affecting performance.
+  virtual bool canUseIntLoadStoreForFloatValues() const { return true; }
```

Comment:
Can you avoid adding a super specific I-want-this-specific-thing-in-one-place style of hook? I would expect this is already expressible in terms of legal operations

Reply:
I'm not sure it can be, the operations are legal (and the codegen is no different from regular AArch64). The issue is the possible performance impact from accessing the same memory region with FP/vector loads/stores and GPR loads/stores in a streaming SVE function.

Reply:
I'm not sure... :sweat_smile: To me, "can" means you _can_ do something if it's beneficial, which using integer stores for FP constants may be. And "should" means you should _always_ do something, but it's not the case that you should always use integer loads for FP values.

Reply:
What about creating a `virtual EVT TargetLowering::getPreferredStoreTypeForConstant(SDNode *C) const;` that returns an appropriate type? This could then also do the check to see if the integer type is legal, for example.